src/DynamoCore/Core/Threading/DynamoScheduler.cs

```diff
@@ -74,6 +74,18 @@ public interface IScheduler
         IEnumerable<AsyncTask> Tasks { get; }
```

Comment:
`IScheduler.Tasks` is not a thread-safe way of exposing such information, I am not sure what else is using this property (and for what purposes), but I'd urge you to update this (existing code) to be this:

```csharp
bool HasPendingTasks
{
    lock(taskQueue)
    {
        return taskQueue.Any();
    }
}
```

This way there will not be race condition when a caller is just about to insert a new task into the queue while this properly is being retrieved. Does that make sense?